src/components/DataTable.tsx

```diff
@@ -269,6 +270,14 @@ export function DataTable<T extends object>(props: Props<T>) {
       });
   };
 
+  const onBulkActionCall = () => {
+    if (mainCheckbox.current) {
+      mainCheckbox.current.checked = false;
+
+      setSelected([]);
+    }
+  };
+
```

Comment:
Can you explain why do we need this one?

Reply:
@beganovich I believe I've covered the function's purpose quite extensively in the comment above, but I'd like to add a few points here as well. I hope my previous comment hit my main idea. Essentially, this function performs the deselection of previously selected entities once the request is completed, regardless of whether it was successful or ended in an error. This approach significantly simplifies the use of bulk actions, particularly when users need to carry out a series of actions successively involving different entities.

Therefore, this function will be passed alongside each `bulkAction` and will be accessible for use within `customBulkAction` hooks. This involves passing it to every callback function within the custom bulk actions hooks:

```
{props.customBulkActions &&
              props.customBulkActions.map(
                (bulkAction: CustomBulkAction<T>, index: number) => (
                  <div key={index}>
                    {bulkAction(selected, selectedResources, onBulkActionCall)}
                  </div>
                )
              )}
```

I did it this way to avoid making things too global. I think there's flexibility for changes. Another option might be using Redux, custom events, or atoms in a similar way. The aim is to make the deselection of selected entities on the table more widespread.

However, I'm quite certain that invoking some form of action to perform deselection under the "finally" block of every bulk action where we want to do deselection is necessary.

What do you think?